src/vs/platform/webContentExtractor/common/webContentExtractor.ts

```diff
@@ -7,12 +7,19 @@ import { URI } from '../../../base/common/uri.js';
 import { createDecorator } from '../../instantiation/common/instantiation.js';
 
 export const IWebContentExtractorService = createDecorator<IWebContentExtractorService>('IWebContentExtractorService');
+export const ISharedWebContentExtractorService = createDecorator<ISharedWebContentExtractorService>('ISharedWebContentExtractorService');
+
 
 export interface IWebContentExtractorService {
 	_serviceBrand: undefined;
 	extract(uri: URI[]): Promise<string[]>;
 }
 
+export interface ISharedWebContentExtractorService {
+	_serviceBrand: undefined;
+	extractUrls(uri: URI): Promise<Uint8Array>;
```

Comment:
Since this is called over IPC, it may be better to return a `VSBuffer` here. I know we optimize how these are sent in some cases

Reply:
Maybe call this `readImage` or something that makes it more clear it's downloading and return data for a single image

Also would be good to have a comment about why it is needed